### PR TITLE
Tasks that can send invitations 

### PR DESCRIPTION
**Pull request by pairing**
**Request specs**
- [x] Request spec for listing pending invitations
- [x] Request spec for accepting invitation
- [x] Request spec for rejecting invitation

**Ember integration tests**
- [x] accept invitation
- [x] reject invitation

**Deployment Notes**

Needs to be merged with: https://github.com/Tahi-project/tahi-standard-cards/pull/1

**Taking note of the following potential refactors: (keeping them here temporarily)**

* dashboard payload serializes redundant data
* Ember data updates may remove need for `litePaper` model
* Paper should really be separated from paper body
* Separating profile from credentials
* Paper should have tasks, instead of through Phases. Phases should be modeled as ordered tags

### CODE REVIEW

**Comment by pairing** [comment]
@mikem Noticed your notes didn't record the person who sent the invite - was that intentional?

File: app/models/invitation.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+class Invitation < ActiveRecord::Base
+  include AASM
+
+  belongs_to :task, inverse_of: :invitations
+  has_one :paper, through: :task
+  belongs_to :invitee, class_name: "User", inverse_of: :invitations
+  belongs_to :actor, class_name: "User", inverse_of: :invitations
+
+  scope :pending, -> { where(state: "pending") }
```
Comment (surfacedamage) [inline app/models/invitation.rb]: This is unnecessary.  AASM gives you these scopes for free.  -AC + CT

File: client/app/pods/dashboard/template.hbs [hunk 2]
```diff
@@ -2,15 +2,25 @@
 {{flash-messages classNames="flash-messages--below-control-bar"}}
 
 <div id="tahi-container">
-  <section id="dashboard-my-submissions">
+
+  <section class="dashboard-section">
+    {{#if pendingInvitations.length}}
+      <h2 class="welcome-message">
+        You have {{model.invitations.length}} {{make-plural pendingInvitations.length "invitation" "invitations"}}.
```
Comment (surfacedamage) [inline client/app/pods/dashboard/template.hbs]: The number and the pluralized word are looking at 2 different things (`model.invitations` vs `pendingInvitations`). It should be consistentificationalized. -- AC + CT

**Comment by surfacedamage** [comment]
@tonywok Looks great with some love on our two comments above!  :+1:

File: app/models/invitation.rb (hunk 1 above)
Comment (tonywok) [inline app/models/invitation.rb]: Ah, you are correct. This was put in before we included `aasm`. Thanks for the :eyes:

**Comment by mikem** [comment]
Regarding storing the person who initiated the invitation: there wasn't an immediate a use case for that. The entity that initiated the invitation is the card on the paper's workflow. Maybe the activity feed will record who triggered an invitation.

Having said all that, it totally makes sense to me to capture the user who initiates the invitation.

**Pull request merged**